src/packet_router.rs

```diff
@@ -0,0 +1,23 @@
+use super::packet;
+use super::packet::Packet;
+use std::net::TcpStream;
+
+mod handshake_init;
+mod client_ping_init;
+mod login_init;
+mod border_cross_login_init;
+mod in_peer_sub_init;
+mod out_peer_sub_init;
+
+// Routes the packet to the corresponding service according to the connection state
+pub fn route_packet(p: Packet, state: &mut u64, stream: &mut TcpStream) {
+    match state {
+        0 => handshake_init::init_handshake(p, state),
```

Comment:
Should we turn this casual enum into an actual enum with like State::Handshake? Seeing numbers in code feels kind of weird but also it doesn't matter that much

Reply:
Yes we could definitely do that

Reply:
Probably fits in this PR, if you have time. Otherwise I can throw it in with my pr